lib/internal/bootstrap/node.js

```diff
@@ -307,6 +304,23 @@ if (process.env.NODE_V8_COVERAGE) {
   };
 }
 
+// Worker threads don't receive signals.
+if (isMainThread) {
+  const {
+    isSignal,
+    startListeningIfSignal,
+    stopListeningIfSignal
+  } = mainThreadSetup.createSignalHandlers();
+  process.on('newListener', startListeningIfSignal);
+  process.on('removeListener', stopListeningIfSignal);
+  // re-arm pre-existing signal event registrations
+  // with this signal wrap capabilities.
+  const signalEvents = process.eventNames().filter(isSignal);
+  for (const ev of signalEvents) {
+    process.emit('newListener', ev);
+  }
+}
+
```

Comment:
Is there any specific reason for placing this block of code at this location? Why I ask is:

 - the re-arming logic was introduced to support `report` - though it supports any code block that registers signals while in the bootstrap process.
 - right now the signal handler setup is established just before the report is established, making the re-arming un-necessary, at least for report.
 - if we place this block of code as early as the process object's creation, we can eliminate the re-arming
 - if we place this block as late as towards the exit of setup, the re-arming will be leveraged by any one in the middle.
 - right now it seems to be a mix of both

and hence my question on the placement - is there any other considerations for this being here?

Code changes looks good to me.